app/models/manageiq/providers/azure/event_catcher_mixin.rb

```diff
@@ -14,6 +14,10 @@ def parse_event_action(event_action)
     "#{object_class}_#{event_type}_"
   end
 
+  def parse_provider_type(event)
+    event["resourceProviderName"]["value"]
+  end
+
```

Comment:
Are we following an interface here? Otherwise this seems unnecessary as a separate method.